src/Storefront/Resources/app/storefront/src/main.js

```diff
@@ -112,17 +112,21 @@ PluginManager.register('DateFormat', DateFormat, '[data-date-format]');
 PluginManager.register('CookiePermission', CookiePermissionPlugin, '[data-cookie-permission]');
 PluginManager.register('CookieConfiguration', CookieConfigurationPlugin, '[data-cookie-permission]');
 PluginManager.register('ScrollUp', ScrollUpPlugin, '[data-scroll-up]');
-PluginManager.register('SearchWidget', SearchWidgetPlugin, '[data-search-form]');
+PluginManager.register('SearchWidget', SearchWidgetPlugin, '[data-search-form]'); /* @deprecated tag:6.5 use data-search-widget instead */
```

Comment:
Hi @JoshuaBehrens,

thanks for the contribution. 👍  Overall I agree we should definitely fix those to match the plugin name.

I'm just curious what would happen when the deprecation is about to be removed one day, because there is actually no change right now in the template where the selector lives. I would prefer a deprecation workflow where really all areas which need to be changed have a deprecation annotation. So basically we can add the same deprecation annotation in the templates where the selectors are. This way it cannot be forgotten and it is more clear that there is also a need to change those in the templates as well.

Since we cannot have both selectors at the same time we could implement the fixed version already behind a feature flag, for example:

```js
/** @deprecated tag:v6.5.0 Registering plugin on selector "data-search-form" is deprecated. Use "data-search-widget" instead */
if (Feature.isActive('NEXT_12345'))
    PluginManager.register('SearchWidget', SearchWidgetPlugin, '[data-search-widget]');
else {
    PluginManager.register('SearchWidget', SearchWidgetPlugin, '[data-search-form]');
}
```
```twig
{# @deprecated tag:v6.5.0 Registering plugin on selector "data-search-form" is deprecated. Use "data-search-widget" instead #}
<form action="{{ path('frontend.search.page') }}"
    method="get"
    {% if feature('NEXT_12345') %}
    data-search-widget="true"
    {% else %}
    data-search-form="true"
    {% endif %}
    data-url="{{ path('frontend.search.suggest') }}?search="
    class="header-search-form">
```
On feature release for the next major version, we could just remove the else cases and deprecations everywhere and it should already work.

Reply:
Thank you for planning a deprecation flow. Most of it is reasonable to me. Can you explain why we can't have both selectors?

Reply:
I amend this to your suggestion. What do we with this match? https://github.com/shopware/platform/blob/d6c853ca3b6b167d61d70e6329f4fcface5c5245/src/Storefront/Resources/app/storefront/src/plugin/offcanvas-filter/offcanvas-filter.plugin.js#L39-L42

Reply:
I also don't like my update hint for javascript. I don't like the import. Do you have a good idea how to approach?

Reply:
> I amend this to your suggestion. What do we with this match?
> 
> https://github.com/shopware/platform/blob/d6c853ca3b6b167d61d70e6329f4fcface5c5245/src/Storefront/Resources/app/storefront/src/plugin/offcanvas-filter/offcanvas-filter.plugin.js#L39-L42

Hi, sorry for my late response.

I think we need to implement the flag here as well to adjust the selector so it matches the new `off-canvas`. Maybe we can do it by toggling the name with a variable for both selector and the Error:

```js
/**
 * @deprecated tag:v6.5.0 Selector "data-offcanvas-filter-content" is deprecated. Use "data-off-canvas-filter-content" instead.
 * The selector will be changed to match plugin selector "data-off-canvas-filter".
 * Remove variable `offCanvasContentAttr` and use `data-off-canvas-filter-content` directly inside `filterContent` variable.
 */
const offCanvasContentAttr =
    Feature.isActive('NEXT_19709') ?
        'data-off-canvas-filter-content' :
        'data-offcanvas-filter-content';

const filterContent = document.querySelector(`[${offCanvasContentAttr}="true"]`);

if (!filterContent) {
    throw Error(`There was no DOM element with the data attribute "${offCanvasContentAttr}".`)
}
```